js/ad_manager_ooyala.js

```diff
@@ -0,0 +1,207 @@
+/**
+ * Ad Manager For Ooyala video as Ads
+ */
+require("../html5-common/js/utils/InitModules/InitOOHazmat.js");
+require("../html5-common/js/utils/constants.js");
+require("../html5-common/js/utils/environment.js");
+
+OO.Ads.manager(function(_, $) {
+  /**
+   * @class OoyalaAdManager
+   * @classDesc The Ooyala Ads Manager class, registered as an ads manager with the ad manager controller.
+   * Controls how Ooyala videos loaded and played as ads while communicating with the ad manager framework.
+   * @public
+   * @property {string} name Name of the OoyalaAdManager, must match what is sent from Backlot and used at the page level
+   * @property {object} _amc An internal reference to the Ad Manager Fratmework
+   * @property {string} _adEmbedCode Keeps track of the embed code of the ooyala ad that is currently playing
+   * @property {string} __movieEmbedCode Keeps track of the embed code of the main video that is currently playing
+   * @property {boolean} _getMainVideoEmbedCode To get the embed code of the main video from the metadata
+   * @property {boolean} _playedAtleastOnce To keep track of the main video play
+   * @property {object} _adProperties Stores the ad data of the ad that is currently being loaded
+   */
+  var OoyalaAdManager = function() {
+    this.name = "ooyala-ads-manager";
+    this.ready = false;
+    this.movieMetadata = {};
+    this.streams = {};
+    var _amc  = null;
+    var _adEmbedCode = "";
+    var _movieEmbedCode ="";
+    var _getMainVideoEmbedCode =true;
+    var _playedAtleastOnce = false;
+    var _adProperties = {};
+    var _urlOpened = false;
+    var _adEnded = false;
+    
+    /**
+     * Called by the Ad Manager Controller. Use this function to initialize, create listeners, and load
+     * remote JS files.
+     * @method AdManager#initialize
```

Comment:
this needs to be \@method OoyalaAdManager#initialize  same goes for all the functions below.